ynr/apps/candidates/tests/test_models.py

```diff
@@ -24,3 +34,94 @@ def test_get_bulk_add_review_url(self):
     def test_get_sopn_view(self):
         url = self.ballot.get_sopn_url()
         assert url == f"/elections/{self.ballot.ballot_paper_id}/sopn/"
+
+
+class TestBallotMethods(TestCase, SingleBallotStatesMixin):
+    def setUp(self):
+        election = self.create_election("Sheffield Local Election")
+        post = self.create_post(post_label="Ecclesall")
+        self.ballot = self.create_ballot(
+            ballot_paper_id="local.sheffield.ecclesall.2021-05-06",
+            election=election,
+            post=post,
+            winner_count=1,
+            candidates_locked=True,
+        )
+
+    def test_uncontested_when_winner_count_same_as_memberships(self):
+        person = PersonFactory(id=79041, name="Andrew Hudson Cowell")
+        party = PartyFactory(ec_id="PP123", name="Social Democratic Party")
+        MembershipFactory(ballot=self.ballot, person=person, party=party)
+        self.assertTrue(self.ballot.uncontested)
+
+    def test_uncontested_when_winner_count_is_less_than_memberships(self):
+        parties = self.create_parties(3)
+        memberships = self.create_memberships(self.ballot, parties)
+        self.assertFalse(self.ballot.uncontested)
+
+    # def test_uncontested_when_winner_count_is_more_than_memberships(self):
+    #     election = self.create_election("Foo Local Election")
+    #     ballot = self.create_ballot(election=self.election, post=post, winner_count=3, candidates_locked=True)
+    #     parties = self.create_parties(3)
+    #     memberships = self.create_memberships(ballot, parties)
+    #     self.assertFalse(self.ballot.uncontested)
+
+
+class TestBallotQuerySetMethods(TestCase, SingleBallotStatesMixin):
+    def test_uncontested_no_candidates(self):
+        election = ElectionFactory()
+        post = self.create_post(post_label="Strensall")
+        ballot = self.create_ballot(
+            ballot_paper_id="local.york.strensall.2019-05-02",
+            election=election,
+            post=post,
+            winner_count=0,
```

Comment:
The winner count should be at least 1? Otherwise I think this will be returned as `uncontested` as 0 seats up and 0 candidates?